src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -489,6 +489,27 @@ void MavlinkReceiver::handle_message_command_both(mavlink_message_t *msg, const
 			send_storage_information(cmd_mavlink.param1 + 0.5f);
 		}
 
+	} else if (cmd_mavlink.command == MAV_CMD_SET_CAMERA_ZOOM) {
+		struct actuator_controls_s actuator_controls = {};
+		actuator_controls.timestamp = hrt_absolute_time();
+
+		for (size_t i = 0; i < 8; i++) {
+			actuator_controls.control[i] = NAN;
+		}
+
+		switch ((int)(cmd_mavlink.param1 + 0.5f)) {
+		case vehicle_command_s::VEHICLE_CAMERA_ZOOM_TYPE_RANGE:
+			actuator_controls.control[actuator_controls_s::INDEX_CAMERA_ZOOM] = cmd_mavlink.param2 / 50.0f - 1.0f;
```

Comment:
Where do these numbers come from? Are they somehow standardized? In the MAVLink specs it just says:
> The range of valid values depend on the zoom type.
https://mavlink.io/en/messages/common.html#MAV_CMD_SET_CAMERA_ZOOM

Reply:
In https://mavlink.io/en/messages/common.html#CAMERA_ZOOM_TYPE it says a value between 0 and 100 is accepted if you have ZOOM_TYPE_RANGE. I map those values to -1 and 1 actuator controls.

Reply:
Oh got it!